src/apps/common/submodules/portListing/portListing.js

```diff
@@ -1048,44 +1042,129 @@ define(function(require) {
+						name: self.portListingGet(['accountNamesById', request.account_id])
+					},
+					amount: _.size(request.numbers),
+					carrier: {
+						winning: _.get(request, 'winning_carrier', self.i18n.active().portListing.misc.unknownCarrier),
+						losing: _.get(request, 'carrier', self.i18n.active().portListing.misc.unknownCarrier)
+					},
+					id: request.id,
+					name: request.name,
+					reference: request.carrier_reference_number,
+					state: request.port_state
+				};
+			});
+		},
+
 		/**
 		 * @param  {Function} args.success
 		 */
 		portListingHelperListPorts: function(args) {
 			var self = this,
-				listAccountPorts = function(callback) {
-					self.portListingRequestListPort({
-						data: {
-							filters: {
-								paginate: false
-							}
-						},
-						success: function(ports) {
-							callback(null, ports);
-						}
-					});
+				listSuspendedAccountPorts = function(callback) {
+					self.portListingRequestByType(callback, 'suspended');
 				},
-				listDescendantsPorts = function(callback) {
-					self.portListingRequestListDescendantsPorts({
-						data: {
-							filters: {
-								paginate: false
-							}
-						},
-						success: function(ports) {
-							callback(null, ports);
-						}
-					});
+				listSuspendedDescendantsPorts = function(callback) {
+					self.portlistingRequestDescendantsByType(callback, 'suspended');
+				},
+				listProgressingAccountPorts = function(callback) {
+					self.portListingRequestByType(callback, 'progressing');
+				},
+				listProgressingDescendantsPorts = function(callback) {
+					self.portlistingRequestDescendantsByType(callback, 'progressing');
 				},
-				parallelRequests = [listAccountPorts];
+				listCompletedAccountPorts = function(callback) {
+					self.portListingRequestByType(callback, 'completed');
+				},
+				listCompletedDescendantsPorts = function(callback) {
+					self.portlistingRequestDescendantsByType(callback, 'completed');
+				},
+				parallelSuspendedRequests = [listSuspendedAccountPorts],
+				parallelSubmittedRequests = [listProgressingAccountPorts, listCompletedAccountPorts];
 
 			if (self.portListingGet('isMonsterApp')) {
-				parallelRequests.push(listDescendantsPorts);
+				parallelSuspendedRequests.push(listSuspendedDescendantsPorts);
+				parallelSubmittedRequests.push(listProgressingDescendantsPorts, listCompletedDescendantsPorts);
 			}
 
-			monster.parallel(parallelRequests, function(err, results) {
-				var portRequests = _
+			monster.parallel({
+				suspendedList: function(callback) {
+					monster.parallel(parallelSuspendedRequests, function(err, portRequests) {
+						callback(null, portRequests);
+					});
+				},
+				submittedList: function(callback) {
+					monster.parallel(parallelSubmittedRequests, function(err, portRequests) {
+						callback(null, portRequests);
+					});
+				}
+			}, function(error, results) {
+				self.portListingSet('accountNamesById', _
 					.chain(results)
+					.map(function(requests) {
```

Comment:
This could be shortened a little bit from:

```javascript
self.portListingSet('accountNamesById', _
	.chain(results)
	.map(function(requests) {
		return _
			.chain(requests)
			.map(function(payload) {
				return _.map(payload, function(account) {
					return {
						id: account.account_id,
						name: account.account_name
					};
				});
			})
			.flatten()
			.transform(function(object, account) {
				object[account.id] = account.name;
			}, {})
			.value();
	})
	.flatten()
	.value()
);
```

To:


```javascript
self.portListingSet('accountNamesById', _
	.flatMap(results, function(requests) {
		return _
			.chain(requests)
			.flatMap(function(payload) {
				return _.map(payload, function(account) {
					return {
						id: account.account_id,
						name: account.account_name
					};
				});
			})
			.transform(function(object, account) {
				object[account.id] = account.name;
			}, {})
			.value();
	})
);
```